src/types/schema.ts

```diff
@@ -7,3 +7,27 @@ export const loginSchema = z.object({
     .email({ message: "Please enter a valid email" }),
   password: z.string().min(1, { message: "Please enter a password" }),
 });
+
+export const signupSchema = z
+  .object({
+    username: z.string().min(1, { message: "Please enter a username" }),
+    email: z
+      .string()
+      .min(1, { message: "Please enter an email" })
+      .email({ message: "Please enter a valid email" }),
+    firstName: z.string().min(1, { message: "Please enter your first name" }),
+    lastName: z.string().min(1, { message: "Please enter your last name" }),
+    password: z
+      .string()
+      .min(1, { message: "Password must be at least 8 characters " }),
```

Comment:
this is wrong, according to your message, min should be 8 and remove trailing space
```suggestion
      .min(8, { message: "Password must be at least 8 characters" }),
```